### DISCUSSION TITLE
Zigbee commands to control TS0601 / _TZE200_dwcarsat Smart air Sensor

### DISCUSSION THREAD

**Discussion by rhk** [arendst/Tasmota#17307] [Q&A]
Hi,

I got one of these Tuya 6-in-1 'Smart air house keeper' that is supposed to publish
temperature 0-60C
humidity 0-99%
co2 0-1000 ppm
voc 0-99.9ppm
formaldehyde 0-10 ug/m3
pm25 0-1000 ug/m3
(linkquality.)

![image](https://user-images.githubusercontent.com/404543/206300583-75197886-1503-424a-8485-b84ef466301a.png)

More info & stuff of the same device here:
https://zigbee.blakadder.com/Tuya_DCR-KQG.html
https://www.zigbee2mqtt.io/devices/TS0601_smart_air_house_keeper.html
https://github.com/dresden-elektronik/deconz-rest-plugin/issues/5889
https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/tuya/air/__init__.py

I run Sonoff ZB Bridge with Tasmota 9.2.0 and this is what it shows me on the front page:
![image](https://user-images.githubusercontent.com/404543/206299224-97e2c65c-4159-4d04-841e-128482477ccb.png)

ZbInfo:
{"ZbInfo":{"0x4D65":{"Device":"0x4D65","IEEEAddr":"0xA4C138A06B42DC2C","ModelId":"TS0601","Manufacturer":"_TZE200_dwcarsat","Endpoints":[1],"Config":["T01"],"TempTarget":0.8,"Reachable":true,"LastSeen":0,"LastSeenEpoch":1670447400,"LinkQuality":118}}}

Console output:
23:23:14 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","EF00/0212":204,"Endpoint":1,"LinkQuality":120}}}
23:23:14 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","EF00/0213":311,"Endpoint":1,"LinkQuality":118}}}
23:23:14 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","0000/FFDF":"��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i","Endpoint":1,"LinkQuality":118}}}
23:23:15 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","0000/FFDF":"��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i","Endpoint":1,"LinkQuality":118}}}
23:23:15 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","0000/FFDF":"��#+i��#+i��#+i��#+i��#+i��#+i��#+i��#+i","Endpoint":1,"LinkQuality":118}}}
23:23:17 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","EF00/0214":4,"Endpoint":1,"LinkQuality":120}}}

23:35:03 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","EF00/0214":3,"Endpoint":1,"LinkQuality":120}}}
23:35:04 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","TuyaBattery":18,"Endpoint":1,"LinkQuality":120}}}
23:35:04 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","EF00/0216":363,"Endpoint":1,"LinkQuality":120}}}
23:35:05 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","TuyaTempTarget":0.7,"Endpoint":1,"LinkQuality":120}}}
23:35:05 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"0x4D65":{"Device":"0x4D65","EF00/0212":207,"Endpoint":1,"LinkQuality":120}}}

At least E00/0212, 0213, 0214, 0216 seems to have values. In addition to these TuyaTempTarget and TuyaBattery values are exposed, which don't seem to make any sense with this device (no temp target can be set nor does it have battery..)

1) Could someone point me, how to help Tasmota ZbBridge to identify this device correctly & output sensible data?
I actually did some quick testing
![image](https://user-images.githubusercontent.com/404543/206306389-a4b20acc-7ff3-490a-a40b-b4f463f476a0.png)
Some guesses:
**TuyaBattery** = Temp? (unstable, value somewhere between 15-24, no decimals shown. Peaks to about 60 when lighting a match about 10cm from the device - definitely doesn't heat up that much)
**0214** = Humid? When blowing into the device, it goes to ~100, however the low value is around 3 and it's not that dry here.. Maybe needs calibration?
**0216** jumped from about 400 to 800 when blowing. pm2?
**0213** jumps from about 300 to 1005 when blowing. Maybe co2?
**TuyaTempTarget** stays between 0.4-0.9 -> This could be formaldehyde (range 0-10)
**0212** about constant 208 - voc still missing, but range should be 0-99,9..

2) How do I use the configuration commands shown at https://www.zigbee2mqtt.io/devices/TS0601_smart_air_house_keeper.html
3) Could someone explain me, how could I be more successful in taking a new devices into use & store the information to the best place to help others / help Tasmota identify the devices automatically / something? This is now my 3rd similar message here, where I've bought something that doesnt' work out of the box with the ZbBridge. What resources should I study to be able to better understand how the devices / Tasmota ZbBridge works? Is this Tasmota Github discussion or Blakadder the best place to save the specs needed to help others looking for the same information?

My earlier similar questions - thanks for everyone who helped!
https://github.com/arendst/Tasmota/discussions/16779 
https://github.com/arendst/Tasmota/discussions/16449

Thanks!

r

**Comment by pickonedev** [comment]
Hi!

Have you managed to use this product with a tasmotized sonoff zigbee bridge?
If yes, can you explain to me as well? 

These are the values I get, but I don't know what these are representing:
"EF00/0202":7
"EF00/0212":238
"EF00/0213":443
"EF00/0214":4
"EF00/0215":24
"EF00/0216":363

This is what shows me into the bridge
![image](https://user-images.githubusercontent.com/34201387/218861954-54ed1bf8-c0f9-4f9b-a755-18c36c669d58.png)



Thank you in advance!

**Reply by rhk** [reply]
That's exactly how far I also got. I get the values through the Tasmotized Sonoff Bridge, but as the bridge doesn't seem to know the device in advance, it shows the raw Zigbee values (EF00/0212":238 etc, instead of CO2:238 or similar). I tried to guess some of them, and that was basically my question.. 

Hope you're able to sort it out & report here. I ended up selling the device and looking for a better one. Haven't found one yet though.

**Reply by sfromis** [reply]
When having issues running older Tasmota versions like 9.2 in this case, the general plan is to upgrade to current, where many fixes have been done. Do not expect suggestions for making outdated versions work, other than to upgrade.

**Reply by pickonedev** [reply]
> When having issues running older Tasmota versions like 9.2 in this case, the general plan is to upgrade to current, where many fixes have been done. Do not expect suggestions for making outdated versions work, other than to upgrade.

It is not about updated, look at what version I use, 12.3.1. And like I already said... When I was using  version 9.x.x, a device of mine, was working perfect, but after the update, I saw only raw values...

**Comment by pickonedev** [comment]
I don't know what is happening with the tasmota firmware, but in the past (like.... version 9.x.x) the values were not raw for a sensor/switch of a climate I use, but after I got higher version 10.x.x everything was turned into raw values. I was using TuyaTempTarget to change the switch of the climate, but after that update, nothing was the same.
Exactly what is happening with this sensor for air quality.
I will find out the values, today or tomorrow, because I will use two devices on two different bridges (one official) and one with tasmota. I will compare the values and find out which is which.

By the way, do you know if there is any command into the tasmota zigbee bridge, which can map the raw to what we need? Like zbname to change the name of the device, something to change EF00/0212 into CO2 ?

**Comment by pickonedev** [comment]
I have matched the values with the raw data

"EF00/0202":7 - PM2.5
"EF00/0212":238 - temp (23,8)
"EF00/0213":443 - hum (44,3)
"EF00/0214":4 - ch2o  value/1000 = 0.004
"EF00/0215":24 - voc  value/1000 = 0.024
"EF00/0216":363 - co2

**Reply by s-hadinger** [reply]
You can use a plugin file to map back these attributes to official Zigbee ones. See https://tasmota.github.io/docs/Zigbee/#advanced-topic-zigbee-device-plugin

**Reply by cristiancs** [reply]
I created a [gist](https://gist.github.com/cristiancs/52507b8d12143001ec52add3a03318bd) based on this references, it shows the temp and humidity on the web gui
